pkg/registry/apis/folders/register.go

```diff
@@ -0,0 +1,157 @@
+		features:   features,
+		namespacer: request.GetNamespaceMapper(cfg),
+		folderSvc:  folderSvc,
+	}
+	apiregistration.RegisterAPI(builder)
+	return builder
+}
+
+func (b *FolderAPIBuilder) GetGroupVersion() schema.GroupVersion {
+	return b.gv
+}
+
+func addKnownTypes(scheme *runtime.Scheme, gv schema.GroupVersion) {
+	scheme.AddKnownTypes(gv,
+		&v0alpha1.Folder{},
+		&v0alpha1.FolderList{},
+		&v0alpha1.FolderInfo{},
+	)
+}
+
+func (b *FolderAPIBuilder) InstallSchema(scheme *runtime.Scheme) error {
+	addKnownTypes(scheme, b.gv)
+
+	// Link this version to the internal representation.
+	// This is used for server-side-apply (PATCH), and avoids the error:
+	//   "no kind is registered for the type"
+	addKnownTypes(scheme, schema.GroupVersion{
+		Group:   b.gv.Group,
+		Version: runtime.APIVersionInternal,
+	})
+
+	// If multiple versions exist, then register conversions from zz_generated.conversion.go
+	// if err := playlist.RegisterConversions(scheme); err != nil {
+	//   return err
+	// }
+	metav1.AddToGroupVersion(scheme, b.gv)
+	return scheme.SetVersionPriority(b.gv)
+}
+
+func (b *FolderAPIBuilder) GetAPIGroupInfo(
+	scheme *runtime.Scheme,
+	codecs serializer.CodecFactory, // pointer?
+	optsGetter generic.RESTOptionsGetter,
+) (*genericapiserver.APIGroupInfo, error) {
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(v0alpha1.GROUP, scheme, metav1.ParameterCodec, codecs)
+
+	strategy := grafanaregistry.NewStrategy(scheme)
+	store := &genericregistry.Store{
+		NewFunc:                   resourceInfo.NewFunc,
+		NewListFunc:               resourceInfo.NewListFunc,
+		PredicateFunc:             grafanaregistry.Matcher,
+		DefaultQualifiedResource:  resourceInfo.GroupResource(),
+		SingularQualifiedResource: resourceInfo.SingularGroupResource(),
+		CreateStrategy:            strategy,
+		UpdateStrategy:            strategy,
+		DeleteStrategy:            strategy,
+	}
+	store.TableConvertor = utils.NewTableConverter(
+		store.DefaultQualifiedResource,
+		[]metav1.TableColumnDefinition{
+			{Name: "Name", Type: "string", Format: "name"},
+			{Name: "Title", Type: "string", Format: "string", Description: "The display name"},
+			{Name: "Parent", Type: "string", Format: "string", Description: "Parent folder UID"},
+		},
+		func(obj any) ([]interface{}, error) {
+			r, ok := obj.(*v0alpha1.Folder)
+			if ok {
+				accessor := kinds.MetaAccessor(r)
+				return []interface{}{
+					r.Name,
+					r.Spec.Title,
+					accessor.GetFolder(),
+				}, nil
+			}
+			return nil, fmt.Errorf("expected resource or info")
+		})
+	legacyStore := &legacyStorage{
+		service:        b.folderSvc,
+		namespacer:     b.namespacer,
+		tableConverter: store.TableConvertor,
+	}
+
+	storage := map[string]rest.Storage{}
+	storage[resourceInfo.StoragePath()] = legacyStore
+	storage[resourceInfo.StoragePath("parents")] = &subParentsREST{b.folderSvc}
+	storage[resourceInfo.StoragePath("children")] = &subChildrenREST{b.folderSvc}
```

Comment:
we could punt on these for now since nested folders won't be enabled -- but also they do not hurt and i think/hope make the long term strategy more clear